sound/soc/sof/topology.c

```diff
@@ -1858,13 +1873,76 @@ static int sof_process_load(struct snd_soc_component *scomp, int index,
 		offset += pdata->size;
 	}
 
+process:
 	process->size = ipc_data_size;
 	swidget->private = process;
 
 	ret = sof_ipc_tx_message(sdev->ipc, process->comp.hdr.cmd, process,
 				 ipc_size, r, sizeof(*r));
-	if (ret >= 0)
+
+	if (ret >= 0 && ipc_data_size)
```

Comment:
@juimonen we should also return is ret < 0 because that means the ipc message has failed. so I think a check for if (ipc_data_size) should suffice here.

Reply:
@ranj063 hmm ok this is dodgy... I'll refine it and remove the goto's to make it clear.

Reply:
ok